crates/ruff_linter/src/rules/flake8_pytest_style/rules/assertion.rs

```diff
@@ -364,9 +367,84 @@ impl Violation for PytestUnittestRaisesAssertion {
 }
 
 /// PT027
-pub(crate) fn unittest_raises_assertion(
+pub(crate) fn unittest_raises_assertion_call(checker: &mut Checker, call: &ast::ExprCall) {
+    // Bindings in `with` statements are handled by `unittest_raises_assertion_with`.
+    if let Stmt::With(ast::StmtWith { items, .. }) = checker.semantic().current_statement() {
+        if items
+            .iter()
+            .any(|item| item.context_expr.range() == call.range && item.optional_vars.is_some())
+        {
+            return;
+        }
+    }
+
+    if let Some(diagnostic) = unittest_raises_assertion(call, [], checker) {
+        checker.diagnostics.push(diagnostic);
+    }
+}
+
+/// PT027
+pub(crate) fn unittest_raises_assertion_binding(
     checker: &Checker,
+    binding: &Binding,
+) -> Option<Diagnostic> {
+    if !matches!(binding.kind, BindingKind::WithItemVar) {
+        return None;
+    }
+
+    let semantic = checker.semantic();
+
+    let Stmt::With(with) = binding.statement(semantic)? else {
+        return None;
+    };
+
+    let Expr::Call(call) = corresponding_context_expr(binding, with)? else {
+        return None;
+    };
+
+    let mut edits = vec![];
+
+    for reference_id in binding.references() {
+        let reference = semantic.reference(reference_id);
+        let node_id = reference.expression_id()?;
+
+        let mut ancestors = semantic.expressions(node_id).skip(1);
+
+        let Expr::Attribute(ast::ExprAttribute { attr, .. }) = ancestors.next()? else {
+            continue;
+        };
+
+        if attr.as_str() == "exception" {
+            edits.push(Edit::range_replacement("value".to_string(), attr.range));
+        }
+    }
+
+    unittest_raises_assertion(call, edits, checker)
+}
+
+#[inline]
```

Comment:
Inline mainly helps with cross-crate calls and it should mainly be used for hot-functions (there's a trade off with increased code size). Inlining more code can also lead to some compiler optimizations to no longer be applied because the function becomes too large. Furthermore, the compiler already applies heuristics to decide when to inline a function. That's why `inline` should only be used where it matters, I don't think it matters here.
```suggestion
```